tests/test_zarr.py

```diff
@@ -549,7 +561,100 @@ def test_metadata_with_trailing_whitespace(
     runtime.stop()
 
     # load from Zarr
-    group = zarr.open(p.video[0].storage.settings.filename)
+    group = zarr.open(p.video[0].storage.settings.uri)
     data = group["0"]
 
     assert data.attrs.asdict() == metadata
+
+
+def test_write_zarr_to_s3(runtime: Runtime, request: pytest.FixtureRequest):
+    if "ZARR_S3_ENDPOINT" not in os.environ:
+        pytest.skip("ZARR_S3_ENDPOINT not set")
+    if "ZARR_S3_BUCKET_NAME" not in os.environ:
+        pytest.skip("ZARR_S3_BUCKET_NAME not set")
+    if "ZARR_S3_ACCESS_KEY_ID" not in os.environ:
+        pytest.skip("ZARR_S3_ACCESS_KEY_ID not set")
+    if "ZARR_S3_SECRET_ACCESS_KEY" not in os.environ:
+        pytest.skip("ZARR_S3_SECRET_ACCESS_KEY not set")
+
+    zarr_s3_endpoint = os.environ["ZARR_S3_ENDPOINT"]
+    zarr_s3_bucket_name = os.environ["ZARR_S3_BUCKET_NAME"]
+    zarr_s3_access_key_id = os.environ["ZARR_S3_ACCESS_KEY_ID"]
+    zarr_s3_secret_access_key = os.environ["ZARR_S3_SECRET_ACCESS_KEY"]
+
+    dm = runtime.device_manager()
+    props = runtime.get_configuration()
```

Comment:
Nit: Are you accessing anything else from props other than the video? Maybe we can clean some of this code by:

```py
video = runtime.get_configuration().video[0]
```

And then replace all `props.video[0]` with `video`.

Reply:
We use `props` to set the configuration later on. But we can do

```python
props = runtime.get_configuration()
video = props.video[0]
```

as a compromise.